File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    let networks = 0
    let visit = Array(n).fill(false)

    for (let i = 0; i < n; i++) {
        if (!visit[i]) {
            networks ++
            isVisit(i)
        }
    }
    
    function isVisit(index) {
        visit[index] = true
        
        const computer = computers[index]
        
        for (let i = 0; i < computer.length; i++) {
            const isConnect = computer[i] === 1 ? true : false;

            if (!visit[i] && isConnect) {
                isVisit(i)
            }
        }        
    }
    return networks
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    networks = 0
    visit = [False] * n

    def dfs(index):
        visit[index] = True
        
        computer_row = computers[index]
        
        for i in range(n):
            if computer_row[i] == 1 and not visit[i]:
                dfs(i)

    for i in range(n):
        if not visit[i]:
            networks += 1
            dfs(i)
    return networks